src/index.ts

```diff
@@ -43,18 +83,28 @@ export function createIPXHandler ({
       if (params.error) {
         return {
           statusCode: 400,
-          body: params.error
+          body: params.error,
+          headers: plainText
         }
       }
       id = params.id
       modifiers = params.modifiers
     }
 
-    const requestHeaders: Record<string, string> = {}
+    const requestHeaders: Record<string, string> = {
+      [SUBREQUEST_HEADER]: '1'
+    }
     const isLocal = !id.startsWith('http://') && !id.startsWith('https://')
     if (isLocal) {
       const url = new URL(event.rawUrl)
       url.pathname = id
+      if (localPrefix && !url.pathname.startsWith(localPrefix)) {
+        return {
+          statusCode: 400,
+          body: 'Invalid source image path',
+          headers: plainText
```

Comment:
I guess for all the spots with plain text headers we always want these responses as text, i.e. no scenario where it would make sense as JSON?